### PR TITLE
return error in `oc set env RESOURCE` when no env args are provided

### PR DESCRIPTION
**Pull request by juanvallejo**
Related BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1366559

The command `oc set RESOURCE` prompts a user that the resource has been
updated even when no other args are provided:

```
$ oc set env dc/database
deploymentconfig "database" updated
```

This patch makes sure that at least one pair of environment variable
arguments are passed before returning a successful message:

```
$ oc set env dc/database
error: at least one environment variable must be provided in the form of
KEY=VALUE
```

cc @fabianofranz

### CODE REVIEW

File: pkg/cmd/cli/cmd/set/env.go [hunk 1]
```diff
@@ -262,6 +262,9 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 	if err != nil {
 		return err
 	}
+	if len(env) == 0 {
```
Comment (liggitt) [inline pkg/cmd/cli/cmd/set/env.go]: Doesn't this break --list?

Comment (juanvallejo) [inline pkg/cmd/cli/cmd/set/env.go]: Hm, it does, I could add a check to make sure `--list` is not set before returning this error

**Comment by juanvallejo** [comment]
@liggitt ptal
Moved the env argument check to the end, before the success message is printed, so that it doesn't break flags.

**Comment by juanvallejo** [comment]
[test]

**Comment by juanvallejo** [comment]
@openshift-bot, the [last build](https://travis-ci.org/openshift/origin/builds/153009955) failed from the following flakes:
 - [make verify](https://travis-ci.org/openshift/origin/jobs/153009957) failed on: https://github.com/openshift/origin/issues/10445

re[test]

**Comment by juanvallejo** [comment]
@openshift-bot, the [last build](https://travis-ci.org/openshift/origin/builds/153009955) failed from the following flakes:
 - [make verify](https://travis-ci.org/openshift/origin/jobs/153009957) failed on: https://github.com/openshift/origin/issues/10445

re[test]

**Comment by fabianofranz** [comment]
LGTM [merge]

**Pull request merged**